### PR TITLE
sensitivity analysis setup

### PR DESCRIPTION
**Pull request by berkserbet**
### Why:
* This function will get sensitivities and baseloads for base locations.

### What:
* This class will have a basic implementation. This function might be changed in the future.

### Checklist
- [x] Finished My Changes
- [ ] Automated Unit Tests

### CODE REVIEW

File: pipelines/rbsa/tasks/find_sensitivities.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+                x = np.append(x,[heat_sens, cool_sens])
+
+                zipcode_loadshapes[enduse] = x
+
+            zipcode_loadshapes.insert(loc=0, column='zipcode', value=zipcode)
+
+            # print(zipcode_loadshapes.head(4))
+
+
+            loadshapes = loadshapes.append(zipcode_loadshapes)
```
Comment (jongoncalves) [inline pipelines/rbsa/tasks/find_sensitivities.py]: Lots of empty lines and the extraneous commented out print statement - please clean this up

File: pipelines/rbsa/tasks/find_sensitivities.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+        return x
+
+    def get_nonsensitive_A(self, weather):
+        """constructs A matrix for non weather sensitive loads
+        """
+
+        A = np.zeros((len(weather.index),48),float)
+
+        ts = datetime.datetime(weather.index[0].year,1,1,0,0,0)
```
Comment (jongoncalves) [inline pipelines/rbsa/tasks/find_sensitivities.py]: Need a single whitespace after the commas

File: pipelines/rbsa/tasks/find_sensitivities.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        dt = datetime.timedelta(hours=1) 
+
+        for h in range(len(weather.index)):
+            A[h][0] = 1
+            hh = h%24
```
Comment (jongoncalves) [inline pipelines/rbsa/tasks/find_sensitivities.py]: whitespace around `%`

File: pipelines/rbsa/tasks/find_sensitivities.py [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+        return A
+
+    def get_A(self, weather):
+        """constructs A matrix for non weather sensitive loads
+        """
+
+        A = np.zeros((len(weather.index),50),float)
+
+        ts = datetime.datetime(weather.index[0].year,1,1,0,0,0)
```
Comment (jongoncalves) [inline pipelines/rbsa/tasks/find_sensitivities.py]: same space comment after the commas

File: pipelines/rbsa/tasks/find_sensitivities.py [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+        dt = datetime.timedelta(hours=1) 
+
+        for h in range(len(weather.index)):
+            A[h][0] = 1
+            hh = h%24
```
Comment (jongoncalves) [inline pipelines/rbsa/tasks/find_sensitivities.py]: same same `%`

File: pipelines/rbsa/tasks/find_sensitivities.py [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+            
+            if weather.index[h].weekday() < 5:
+                A[h][hh] = 1.0
+            else:
+                A[h][hh+24] = 1.0
+
+            if weather['Temperature'][h] < self.theat:
+                A[h][(24*2)] = weather['Temperature'][h]-self.theat
+            elif weather['Temperature'][h] > self.tcool:
+                A[h][(24*2)+1] = weather['Temperature'][h]-self.tcool
```
Comment (jongoncalves) [inline pipelines/rbsa/tasks/find_sensitivities.py]: same whitespace issue around all the operators above

File: local_data/noaa_weather_cleaner.ipynb [hunk 7]
```diff
@@ -0,0 +1,115 @@
+{
```
[the remark this answers is not in the document]
Comment (berkserbet) [reply local_data/noaa_weather_cleaner.ipynb]: Accidentally added it, removed.

File: pipelines/rbsa/tasks/find_sensitivities.py [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+            { 'name': 'noaa/989.csv', 'read_type': SupportedFileReadType.DATA },
+            { 'name': 'noaa/990.csv', 'read_type': SupportedFileReadType.DATA },
+            { 'name': 'noaa/991.csv', 'read_type': SupportedFileReadType.DATA },
+            { 'name': 'noaa/992.csv', 'read_type': SupportedFileReadType.DATA },
+            { 'name': 'noaa/993.csv', 'read_type': SupportedFileReadType.DATA },
+        ]
+        self.task_function = self._task
+
+        self.theat = 15
+        self.tcool = 25
```
[the remark this answers is not in the document]
Comment (berkserbet) [reply pipelines/rbsa/tasks/find_sensitivities.py]: Good call, putting in config.

File: pipelines/rbsa/tasks/find_sensitivities.py (hunk 1 above)
Comment (berkserbet) [reply pipelines/rbsa/tasks/find_sensitivities.py]: My bad, cleaned.

File: pipelines/rbsa/tasks/find_sensitivities.py (hunk 2 above)
Comment (berkserbet) [reply pipelines/rbsa/tasks/find_sensitivities.py]: fixed

File: pipelines/rbsa/tasks/find_sensitivities.py (hunk 3 above)
Comment (berkserbet) [reply pipelines/rbsa/tasks/find_sensitivities.py]: fixed

File: pipelines/rbsa/tasks/find_sensitivities.py (hunk 4 above)
Comment (berkserbet) [reply pipelines/rbsa/tasks/find_sensitivities.py]: fixed

File: pipelines/rbsa/tasks/find_sensitivities.py (hunk 5 above)
Comment (berkserbet) [reply pipelines/rbsa/tasks/find_sensitivities.py]: fixed

File: pipelines/rbsa/tasks/find_sensitivities.py (hunk 6 above)
Comment (berkserbet) [reply pipelines/rbsa/tasks/find_sensitivities.py]: fixed

File: pipelines/rbsa/tasks/find_sensitivities.py [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+        
+    def _task(self):
+        data_map = self._get_data()
+        
+        self.sensitivity_temperatures = data_map[self.input_artifact_sensitivity_temperatures]['residential']
+
+        self.theat = self.sensitivity_temperatures['theat']
+        self.tcool = self.sensitivity_temperatures['tcool']
+
+        print(self.theat)
```
Comment (jongoncalves) [inline pipelines/rbsa/tasks/find_sensitivities.py]: If this is needed, make it a logging statement.

File: pipelines/rbsa/tasks/find_sensitivities.py [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+                    A_base = A[:,:-2]
+                    x = self.get_baseload(enduse_df, A_base)
+                    x = np.append(x, [0])
+                    x = np.append(x, [0])
+
+                heat_sens = x[-2]
+                cool_sens = x[-1]
+
+                # adjust x by first value
+                x = np.append(x[0],x[1:48]+x[0])
```
Comment (jongoncalves) [inline pipelines/rbsa/tasks/find_sensitivities.py]: space around `+` and after the comma

File: pipelines/rbsa/tasks/find_sensitivities.py [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+        self.on_complete({self.output_artifact_loadshapes: loadshapes})
+
+    def get_baseload(self, df, A):
+        """ This function gets baseload for non weather sensitive enduses
+        """
+
+        At = A.transpose()
+        y = np.asarray(df).transpose()
+        M = np.matmul(np.linalg.inv(np.matmul(At,A)),At)
+        x = np.matmul(M,y)
```
Comment (jongoncalves) [inline pipelines/rbsa/tasks/find_sensitivities.py]: spaces after the above commas

File: config/SENSITIVITY_TEMPERATURES.json [hunk 12]
```diff
@@ -0,0 +1,9 @@
+{
+	"version": 1,
+	"application": "loadinsight",
+	"name": "SENSITIVITY_TEMPERATURES",
+	"residential" : {
+		"theat":15,
+		"tcool":25
+	}
```
Comment (jongoncalves) [inline config/SENSITIVITY_TEMPERATURES.json]: remove the whitespace before the colon after `resedential` and add space after the colon on `theat` and `tcool`

File: pipelines/rbsa/tasks/find_sensitivities.py (hunk 9 above)
Comment (berkserbet) [reply pipelines/rbsa/tasks/find_sensitivities.py]: not needed

File: pipelines/rbsa/tasks/find_sensitivities.py (hunk 10 above)
Comment (berkserbet) [reply pipelines/rbsa/tasks/find_sensitivities.py]: fixed

File: pipelines/rbsa/tasks/find_sensitivities.py (hunk 11 above)
Comment (berkserbet) [reply pipelines/rbsa/tasks/find_sensitivities.py]: fixed

File: config/SENSITIVITY_TEMPERATURES.json (hunk 12 above)
Comment (berkserbet) [reply config/SENSITIVITY_TEMPERATURES.json]: fixed

**Pull request merged**